src/libs/ajax/workspaces/workspace-models.ts

```diff
@@ -30,3 +30,102 @@ export interface MethodConfiguration {
   deleted?: boolean;
   dataReferenceName?: string;
 }
+
+// TYPES RELATED TO WORKSPACE SETTINGS
+export type WorkspaceSetting = BucketLifecycleSetting | SoftDeleteSetting;
```

Comment:
Before this was more lenient, but I decided to limit it to the known types (though I expect others will be added). This change did impact the tests though.

Previously:
```
export interface WorkspaceSetting {
  settingType: string;
}